### PR TITLE
Frodriguezhervias/agora 1506 fix iscitizen query and filter by mostvotingpower on

### PR DESCRIPTION
**Pull request by ferrodri**
Solves https://linear.app/agora-app/issue/AGORA-1506/fix-iscitizen-query-and-filter-by-mostvotingpower-on-citizens-tab
Preview link: https://agora-next-hgqnjlrhh-voteagora.vercel.app/

### CODE REVIEW

File: src/app/api/delegates/getDelegates.ts [hunk 1]
```diff
@@ -62,19 +62,23 @@ export async function getDelegates({
...
   );
 
   return {
     meta,
     delegates: delegates.map((delegate, index) => ({
       address: delegate.delegate,
       votingPower: delegate.voting_power?.toFixed(0),
-      statement: statements[index],
-    })),
+      citizen: _delegates[index].citizen.length > 0,
```
Comment (kent) [inline src/app/api/delegates/getDelegates.ts:79]: Why this and not boolean check?

Comment (kent) [reply src/app/api/delegates/getDelegates.ts:79]: Gotcha. I think this is fine, something has me worried but we might want to do something with that metadata.

**Review by kent — APPROVED** [review]
LGTM

**Comment by kent** [comment]
@stepandel I had a few questions for Fernando but would love for you to look at the getDelegates method here and make sure you are good with it.

File: src/app/api/citizens/getCitizens.ts [hunk 2]
```diff
@@ -16,41 +24,51 @@ export async function getCitizens({
...
+        return prisma.$queryRaw<citizen[]>(
+          Prisma.sql`
+            SELECT address_metadata.address, address_metadata.metadata, delegate.voting_power
+            FROM center.address_metadata address_metadata
+            JOIN center.delegates delegate ON LOWER(address_metadata.address) = LOWER(delegate.delegate)
+            WHERE address_metadata.kind = 'citizen' 
+            AND address_metadata.dao_slug = 'OP'
+            ORDER BY delegate.voting_power DESC
+            OFFSET ${skip}
+            LIMIT ${take};
```
Comment (stepandel) [inline src/app/api/citizens/getCitizens.ts:53]: love this!

File: src/app/api/delegates/getDelegates.ts [hunk 3]
```diff
@@ -62,19 +62,23 @@ export async function getDelegates({
...
   );
 
-  const statements = await Promise.all(
-    delegates.map((delegate) =>
-      getDelegateStatement({ addressOrENSName: delegate.delegate })
-    )
+  const _delegates = await Promise.all(
+    delegates.map(async (delegate) => {
+      return {
+        citizen: await isCitizen(delegate.delegate),
```
Comment (stepandel) [inline src/app/api/delegates/getDelegates.ts:68]: out of curiosity, wondering if you tried to throw this directly in the query?

it's still very performant, so don't think there's a need

**Review by stepandel — APPROVED** [review]
Nice, good job!

**Pull request merged**